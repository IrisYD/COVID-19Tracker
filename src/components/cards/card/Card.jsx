import React from "react";
import {Card, CardContent, Typography, Grid} from "@mui/material";
// import CountUp from "react-countup";
import cx from "classnames";
import styles from "./Card.css";

const CovidCard = ({className, cardTitle, value, lastUpdate, cardSubtitle}) => (
    <Grid item xs={12} mid={3} component={Card} className={cx(styles.card, className)}>
        <CardContent>
            <Typography color="textPrimary" gutterBottom>
                {cardTitle}
            </Typography>
            <Typography variant="h5" component="h2">
                {/*<CountUp start={0} end={value} duration={2.75} separator="." />*/}
            </Typography>
            <Typography color="textSecondary">
                {
                    new Date(lastUpdate).toDateString()
                }
            </Typography>
            <Typography variant="body2" component="p">
                {cardSubtitle}
            </Typography>
        </CardContent>
    </Grid>
);

export default CovidCard;
