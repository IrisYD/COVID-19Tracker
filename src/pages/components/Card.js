import React from 'react';


function Card(props) {
    const { user, post } = props;
    // const { username } = React.useContext(AppContext);

    const avatars = ['../images/spiderman.png', '../images/batman.png', '../images/wonderwomen.png', '../images/ironman.png'];

    const random = Math.floor(Math.random() * avatars.length);

    return <li>
        <div className='user'>
            <div className='avatar' style={{ backgroundImage: `url(${avatars[random]})` }} />
            <div className='user_name'><b>{post.userName || "unknown"}</b></div>
            <div>Age: {post.userAge}</div>
            {/* <div>No underlying disease</div> */}
        </div>
        <div className='info'>
            <div className='row'>
                <b><h4>{post.symptoms.map((symptom) => (
                    <span key={symptom}>{symptom}</span>
                ))}</h4></b>
            </div>
            <div className='row'>
            <span>Vaccine Taken:   <b>{post.vaccine ? post.vaccine : post.userVaccineBrand}</b></span>
            </div>
            <div className='row'>
            <span>Vaccine Status:   <b>{post.vaccineStatus ? post.vaccineStatus : post.userVaccineStatus}</b></span>
            </div>
            <div className='row'>
            <span>Test Result:   <b>{post.testResult ? post.testResult : "No Test Result"}</b></span>
            </div>
            <div className='row'>
                <span>Start To Show Symptoms:   <b>{post.startTime ? post.startTime : 'User did not choose the Start Time'}</b></span>
            </div>
            <div className='row comments-row'>
                <p>
                    {post?.comments}
                </p>
                <div className='comments'>
                    {/* {user?.age} */}
                </div>
            </div>
        </div>
    </li>
}

export default Card;