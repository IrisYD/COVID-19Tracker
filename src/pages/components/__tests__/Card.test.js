import Card from '../Card';
import { render } from '@testing-library/react';

const defaultPost = {
    comments: "122sdfasdfds",
    startTime: "02/03/2022",
    symptoms: ['Cough'],
    testResult: "positive",
    vaccine: "moderna",
    _id: "6243ba37cd7be3119f69513a"
}

xtest("should render", () => {
    const { container } = render(<Card post={defaultPost} />);
    expect(container).toMatchSnapshot();
});