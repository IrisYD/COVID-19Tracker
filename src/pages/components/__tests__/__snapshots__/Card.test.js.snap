// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test should render 1`] = `
<div>
  <li>
    <div
      class="user"
    >
      <div
        class="avatar"
        style="background-image: url(../images/spiderman.png);"
      />
      <div
        class="user_name"
      >
        <b>
          unknown
        </b>
      </div>
      <div>
        Age: 
      </div>
    </div>
    <div
      class="info"
    >
      <div
        class="row"
      >
        <b>
          <h4>
            <span>
              Cough
            </span>
          </h4>
        </b>
      </div>
      <div
        class="row"
      >
        <span>
          Vaccine Taken:   
          <b>
            moderna
          </b>
        </span>
      </div>
      <div
        class="row"
      >
        <span>
          Vaccine Status:   
          <b />
        </span>
      </div>
      <div
        class="row"
      >
        <span>
          Test Result:   
          <b>
            positive
          </b>
        </span>
      </div>
      <div
        class="row"
      >
        <span>
          Start To Show Symptoms:   
          <b>
            02/03/2022
          </b>
        </span>
      </div>
      <div
        class="row comments-row"
      >
        <p>
          122sdfasdfds
        </p>
        <div
          class="comments"
        />
      </div>
    </div>
  </li>
</div>
`;

exports[`should render 1`] = `
<div>
  <li>
    <div
      class="user"
    >
      <div
        class="avatar"
        style="background-image: url(../images/spiderman.png);"
      />
      <div
        class="user_name"
      >
        <b>
          unknown
        </b>
      </div>
      <div>
        Age: 
      </div>
    </div>
    <div
      class="info"
    >
      <div
        class="row"
      >
        <b>
          <h4>
            <span>
              Cough
            </span>
          </h4>
        </b>
      </div>
      <div
        class="row"
      >
        <span>
          Vaccine Taken:   
          <b>
            moderna
          </b>
        </span>
      </div>
      <div
        class="row"
      >
        <span>
          Vaccine Status:   
          <b />
        </span>
      </div>
      <div
        class="row"
      >
        <span>
          Test Result:   
          <b>
            positive
          </b>
        </span>
      </div>
      <div
        class="row"
      >
        <span>
          Start To Show Symptoms:   
          <b>
            02/03/2022
          </b>
        </span>
      </div>
      <div
        class="row comments-row"
      >
        <p>
          122sdfasdfds
        </p>
        <div
          class="comments"
        />
      </div>
    </div>
  </li>
</div>
`;
